staging/src/k8s.io/apiserver/pkg/storage/selection_predicate.go

```diff
@@ -128,3 +129,21 @@ func (s *SelectionPredicate) MatchesSingle() (string, bool) {
 func (s *SelectionPredicate) Empty() bool {
 	return s.Label.Empty() && s.Field.Empty()
 }
+
+// For any index defined by IndexFields, if a matcher can match only (a subset)
+// of objects that return <value> for a given index, a pair (<index name>, <value>)
+// wil be returned.
+func (s *SelectionPredicate) MatcherIndex() []MatchValue {
+	var result []MatchValue
+	for _, field := range s.IndexFields {
+		if value, ok := s.Field.RequiresExactMatch(field); ok {
+			result = append(result, MatchValue{IndexName: field, Value: value})
+		}
+	}
+	for _, label := range s.IndexLabels {
+		if value, ok := s.Label.RequiresExactMatch(label); ok {
+			result = append(result, MatchValue{IndexName: label, Value: value})
```

Comment:
What prevents field names from colliding with label names, given that you use both as the IndexName?

Reply:
Indeed， the indxer user(e.g pod storage) should guarantee not use the same indexName

Reply:
I think the indexing system needs make this mistake impossible; otherwise it's a new constraint developers have to keep in mind over a larger swath of the code base.